### PR TITLE
a-tiny-js-world initial commit on 5th of January

### PR DESCRIPTION
**Pull request by nataliereshetnikova**
### A tiny js world

[Demo](https://nataliereshetnikova.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/nataliereshetnikova/a-tiny-JS-world/blob/gh-pages/index.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@nataliereshetnikova There is a couple of methods to let a reviewer an idea that you are ready for the next review iteration:
- resolve conversations you believe you elaborated on
- tell explicitly

Please, note that a new commit is not a sufficient evidence per se as it is not a rare case when changes get implemented in more than one commit.

So it is good to have a clear understanding if PR author has things done.

File: submissions/nataliereshetnikova/a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      let names = [];
+      item.forEach((frnd) => {
+        names.push('friend '+frnd.name);
+      });
+      arr.push(names.join('; '));
+    }else{
+      arr.push(item);
+    }
+  });
+  return arr.join('; ');
```
Comment (OleksiyRudenko) [inline submissions/nataliereshetnikova/a-tiny-js-world/index.js]: Order of properties in an object is not guaranteed. You past version was more appropriate and just needed some improvement.
Hint: 
```
[
  varA,
  "string",
  a.map().join(),
].join();
```
``

File: submissions/nataliereshetnikova/a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name: 'Andrew',
+  gender: 'male',
+  legs: 2,
+  hands: 2,
+  saying: 'Hi!'
+};
+const woman = JSON.parse(JSON.stringify(man));
+woman.name = 'Natalie';
+woman.gender = 'female';
+woman.saying = 'Nice of you!';
```
[the remark this answers is not in the document]
Comment (nataliereshetnikova) [reply submissions/nataliereshetnikova/a-tiny-js-world/index.js]: Thank you for review! While using Object.assign, the copied object then changes the original one, so I guess I couldn't use it here

File: submissions/nataliereshetnikova/a-tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const cat = JSON.parse(JSON.stringify(dog));
+  cat.species = 'cat';
+  cat.name = 'Kitt';
+  cat.gender = 'male';
+  cat.saying='meow!';
+man.friends = [dog.name, woman.name];
+woman.friends = [man.name, cat.name];
+print(dog.species + ';' + dog.name + ';' + dog.gender + ';' +
+  dog.legs + ';' + dog.hands + ';' + dog.saying + ';');
+print(cat.species + ';' + cat.name + ';' + cat.gender + ';' +
```
[the remark this answers is not in the document]
Comment (nataliereshetnikova) [reply submissions/nataliereshetnikova/a-tiny-js-world/index.js]: Thank you, I've used now a function and Array,join, which looks way nicer than previously

**Comment by OleksiyRudenko** [comment]
@nataliereshetnikova Do you need any support or clarifications on the above?

**Comment by zonzujiro** [comment]
Looks like author forgot about PR. I will close it.

**Pull request closed without merging**